db/schema.rb

```diff
@@ -13,39 +12,39 @@
 
 ActiveRecord::Schema.define(version: 20130803071120) do
 
-  create_table "admins", force: true do |t|
-    t.string   "password_hash"
-    t.string   "password_salt"
-    t.datetime "created_at"
-    t.datetime "updated_at"
+  create_table "admins", force: :cascade do |t|
+    t.string "password_hash"
+    t.string "password_salt"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
   end
 
-  create_table "boards", force: true do |t|
-    t.string   "name"
-    t.datetime "created_at"
-    t.datetime "updated_at"
-    t.string   "banner_file_name"
-    t.string   "banner_content_type"
-    t.integer  "banner_file_size"
+  create_table "boards", force: :cascade do |t|
+    t.string "name"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.string "banner_file_name"
+    t.string "banner_content_type"
+    t.integer "banner_file_size"
     t.datetime "banner_updated_at"
   end
 
-  create_table "branches", force: true do |t|
+  create_table "branches", force: :cascade do |t|
     t.datetime "last_post_at"
-    t.datetime "created_at"
-    t.datetime "updated_at"
-    t.integer  "board_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer "board_id"
   end
 
-  create_table "leafs", force: true do |t|
-    t.string   "name"
-    t.text     "content"
-    t.datetime "created_at"
-    t.datetime "updated_at"
-    t.integer  "branch_id"
-    t.string   "photo_file_name"
-    t.string   "photo_content_type"
-    t.integer  "photo_file_size"
+  create_table "leafs", force: :cascade do |t|
+    t.string "name"
+    t.text "content"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
```

Comment:
This *shouldnt* break production, but if there are any records with NULL created_at/updated_at fields the migration will fail, I'm pretty sure.

Reply:
The alternative is to mark all the timestamp fields `null: true` in the migration files. This change was sometime in the rails 5.x series I believe.

Reply:
Yeah..  i didn't do any of this, ... was just generated automatically for me. idk if it was the `rake db:migrate` or `rake setup`... kinda rusty with rails.

Reply:
that said i would be surprised if there are any Leafs on prod with an empty `created_at` or `updated_at`. I guess we can just check.

Reply:
I think it probably happened when you ran `migrate` if you were starting on a fesh database. Its probably best to use `null: true` for the previous migrations. 

So we should add `null: true` option to all datetime fields in the old migrations.

Then we can create a new migration for changing the columns to `null: false` if we want to.

I mean, at least that's what I would do if this was a project at a company...maybe it doesn't make sense for an open source project.